src/storage/kv/mod.rs

```diff
@@ -313,6 +314,32 @@ pub unsafe fn destroy_tls_engine<E: Engine>() {
     });
 }
 
+/// Get a snapshot of `engine`.
+pub fn snapshot<E: Engine>(
+    engine: &E,
+    ctx: &Context,
+) -> impl std::future::Future<Output = Result<E::Snap>> {
+    let (callback, future) =
+        tikv_util::future::paired_must_called_std_future_callback(drop_snapshot_callback::<E>);
+    let val = engine.async_snapshot(ctx, callback);
+    // make engine not cross yield point
+    async move {
+        val?; // propagate error
+        let (_ctx, result) = future
+            .map_err(|cancel| Error::from(ErrorInner::Other(box_err!(cancel))))
+            .await?;
+        result
+    }
+}
+
+pub fn drop_snapshot_callback<E: Engine>() -> (CbContext, Result<E::Snap>) {
+    let bt = backtrace::Backtrace::new();
```

Comment:
Is backtrace necessary? With the error we can know the line number about the drop occurs.

Reply:
Either the log macro or the `box_err` macro only shows where the macro is called (so that will be in the `drop_snapshot_callback` method). This doesn't provide much information.